File: SuperiorPackGroup/DAL/DALUtils.vb
Imports System.Data.Sql
Imports System.Data.SqlClient
Imports System.Collections.Specialized
Imports DevExpress.Xpo
Imports DXDAL.SPGData

Public Class DALUtils

    Public Shared Function TestConnection() As Boolean

        Return TestConnection(Nothing)

    End Function

    Public Shared Function TestConnection(ByVal connectionString As String) As Boolean

        If String.IsNullOrEmpty(connectionString) Then
            If String.IsNullOrEmpty(My.Settings.UserConnectionString) Then
                Return False
            Else
                connectionString = My.Settings.UserConnectionString
            End If
        End If

        Dim conn As SqlConnection = New SqlConnection(connectionString)

        Try
            conn.Open()
            If Session.DefaultSession.IsConnected Then
                Session.DefaultSession.Disconnect()
            End If
            Session.DefaultSession.ConnectionString = connectionString
            Session.DefaultSession.AutoCreateOption = DB.AutoCreateOption.SchemaOnly
            Session.DefaultSession.UpdateSchema(Reflection.Assembly.GetAssembly(GetType(LatestVersion)))
            SPGDataLayer = XpoDefault.GetDataLayer(connectionString, DB.AutoCreateOption.SchemaOnly)
            Return True
        Catch ex As Exception
            Return False
        Finally
            conn.Close()
        End Try

    End Function

    Public Shared Function GetSQLServers() As ArrayList

        Dim sqlSource As SqlDataSourceEnumerator = SqlDataSourceEnumerator.Instance
        Dim dt As DataTable = sqlSource.GetDataSources()
        Dim serverList As ArrayList = New ArrayList()

        For Each dr As DataRow In dt.Rows
            If String.IsNullOrEmpty(dr("InstanceName").ToString) OrElse String.Compare(dr("InstanceName").ToString, "MSSQLSERVER") = 0 Then
                serverList.Add(dr("ServerName").ToString())
            Else
                serverList.Add(String.Format("{0}\{1}", dr("ServerName").ToString(), dr("InstanceName").ToString()))
            End If
        Next

        Return serverList

    End Function

    Public Shared Function GetDatabases(ByVal connectionString As String) As ArrayList

        If String.IsNullOrEmpty(connectionString) Then
            If String.IsNullOrEmpty(My.Settings.UserConnectionString) Then
                Return Nothing
            Else
                connectionString = My.Settings.UserConnectionString
            End If
        End If

        Dim conn As SqlConnection = New SqlConnection(connectionString)
        Dim da As SqlDataAdapter = New SqlDataAdapter("sp_helpdb", conn)
        Dim ds As DataSet = New DataSet()
        Dim dataBaseList As ArrayList = New ArrayList()
        Try
            da.Fill(ds)

            For Each dr As DataRow In ds.Tables(0).Rows
                dataBaseList.Add(dr("Name").ToString())
            Next

            Return dataBaseList
        Catch ex As Exception
            MessageBox.Show(ex.Message)
            Return Nothing
        End Try

    End Function

End Class

File: SuperiorPackGroup/DAL/DALUtils.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

using System.Data.Sql;
using System.Data.SqlClient;
using System.Collections.Specialized;
using DevExpress.Xpo;
using DXDAL.SPGData;

namespace SuperiorPackGroup
{
	public class DALUtils
	{

		public static bool TestConnection()
		{

			return TestConnection(null);

		}

		public static bool TestConnection(string connectionString)
		{

			if (string.IsNullOrEmpty(connectionString))
			{
				if (string.IsNullOrEmpty(Properties.Settings.Default.UserConnectionString))
				{
					return false;
				}
				else
				{
					connectionString = Properties.Settings.Default.UserConnectionString;
				}
			}

			SqlConnection conn = new SqlConnection(connectionString);

			try
			{
				conn.Open();
				if (Session.DefaultSession.IsConnected)
				{
					Session.DefaultSession.Disconnect();
				}
				Session.DefaultSession.ConnectionString = connectionString;
				Session.DefaultSession.AutoCreateOption = DevExpress.Xpo.DB.AutoCreateOption.SchemaOnly;
				Session.DefaultSession.UpdateSchema(System.Reflection.Assembly.GetAssembly(typeof(LatestVersion)));
				MyDataLayers.SPGDataLayer = XpoDefault.GetDataLayer(connectionString, DevExpress.Xpo.DB.AutoCreateOption.SchemaOnly);
				return true;
			}
			catch (Exception ex)
			{
				return false;
			}
			finally
			{
				conn.Close();
			}

		}

		public static ArrayList GetSQLServers()
		{

			SqlDataSourceEnumerator sqlSource = SqlDataSourceEnumerator.Instance;
			DataTable dt = sqlSource.GetDataSources();
			ArrayList serverList = new ArrayList();

			foreach (DataRow dr in dt.Rows)
			{
				if (string.IsNullOrEmpty(dr["InstanceName"].ToString()) || string.Compare(dr["InstanceName"].ToString(), "MSSQLSERVER") == 0)
				{
					serverList.Add(dr["ServerName"].ToString());
				}
				else
				{
					serverList.Add(string.Format("{0}\\{1}", dr["ServerName"].ToString(), dr["InstanceName"].ToString()));
				}
			}

			return serverList;

		}

		public static ArrayList GetDatabases(string connectionString)
		{

			if (string.IsNullOrEmpty(connectionString))
			{
				if (string.IsNullOrEmpty(Properties.Settings.Default.UserConnectionString))
				{
					return null;
				}
				else
				{
					connectionString = Properties.Settings.Default.UserConnectionString;
				}
			}

			SqlConnection conn = new SqlConnection(connectionString);
			SqlDataAdapter da = new SqlDataAdapter("sp_helpdb", conn);
			DataSet ds = new DataSet();
			ArrayList dataBaseList = new ArrayList();
			try
			{
				da.Fill(ds);

				foreach (DataRow dr in ds.Tables[0].Rows)
				{
					dataBaseList.Add(dr["Name"].ToString());
				}

				return dataBaseList;
			}
			catch (Exception ex)
			{
				MessageBox.Show(ex.Message);
				return null;
			}

		}

	}

}